### PR TITLE
docs: Add notes on how to do a new release.

### PR DESCRIPTION
**Pull request by randomvariable**
Notes on how to do a release.

### CODE REVIEW

File: docs/releasing.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Release process
+
+## Manual
+
+1. Login using `gcloud auth login`. You should see
+   `cluster-api-provider-vsphere` in your list of available projects.
+2. Make sure your repo is clean by git's standards
+3. If this is a new minor release, create a new release branch and push
+   to github, otherwise switch to it, for example `release-0.7`
+4. Create an env var VERSION=0.x.x Ensure the version is prefixed with a v
```
Comment (ncdc) [inline docs/releasing.md:10]:
```suggestion
4. Create an env var VERSION=v0.x.x Ensure the version is prefixed with a v
```

File: docs/releasing.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+5. Tag the repository and push the tag `git tag -s -m $VERSION $VERSION`
```
Comment (ncdc) [inline docs/releasing.md:11]: If it's worth it, CAPI mentions both annotated and GPG signed tag options (https://github.com/kubernetes-sigs/cluster-api/blob/23aad1f1e498f564d7b745def25930ecd62c615f/docs/developer/releasing.md):
>1. Create an annotated tag `git tag -a v0.x.y -m v0.x.y`
>    1. To use your GPG signature when pushing the tag, use `git tag -s [...]` instead

File: docs/releasing.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+6. Push the tag using git push upstream $VERSION
```
Comment (ncdc) [inline docs/releasing.md:12]:
```suggestion
6. Push the tag using `git push upstream $VERSION`
```

File: docs/releasing.md (hunk 2 above)
Comment (randomvariable) [reply docs/releasing.md]: Updated

**Pull request merged**